### PR TITLE
Regions

### PR DESCRIPTION
**Pull request by revol-xut**
# Project

This Pull Request implements features discussed here #2 

### Things Done
- **POST /region** create
- **GET /region** list
- **GET /region/id** info
- **PUT /region/id** update
- **DELETE /region/id** delete or deactivate

### Checklist

- [x] rust formatting 
- [x] added to utoipa for open api-spec
- [x] small docs annotations

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -75,21 +73,33 @@ async fn main() -> std::io::Result<()> {
...
             .route("/user/list", web::get().to(routes::user::user_list))
+            .route(
+                "/region/create",
+                web::get().to(routes::region::region_create),
+            )
+            .route("/region/list", web::get().to(routes::region::region_list))
+            .route(
+                "/region/update",
+                web::get().to(routes::region::region_update),
+            )
```
Comment (MarcelCoding) [inline src/main.rs:102]: Shouldn't'd be this something like this?
```suggestion
            .route(
                "/region",
                web::post().to(routes::region::region_create),
            )
            .route("/region", web::get().to(routes::region::region_list))
            .route(
                "/region/:id", // i don't know how path params in artix work
                web::put().to(routes::region::region_update),
            )
```

Comment (revol-xut) [reply src/main.rs]: totally correct fixed it

**Comment by revol-xut** [comment]
Other question, do we want region delete ? Its harmless as long as there are no stations with this region or received data.

**Comment by MarcelCoding** [comment]
Like they say at my company, we don't delete, we just deactivate.

**Comment by revol-xut** [comment]
Yes we choose the same approach, barbecue deleting would introduce inconsistencies that you need to take then into consideration.

**Pull request merged**